src/backends/sentry_backend_crashpad.cpp

```diff
@@ -439,9 +439,17 @@ crashpad_backend_startup(
     if (minidump_url) {
         SENTRY_DEBUGF("using minidump URL \"%s\"", minidump_url);
     }
+    auto proxy_url = "";
+    if (strncmp(options->dsn->raw, "https", 5) == 0) {
+        proxy_url = getenv("https_proxy");
+    } else if (strncmp(options->dsn->raw, "http", 4) == 0) {
+        proxy_url = getenv("http_proxy");
+    }
+    proxy_url = options->proxy ? options->proxy
+        : proxy_url != NULL    ? proxy_url
+                               : "";
```

Comment:
Do we need a test for this?

Reply:
yes, we want to test the following:
- setting `https_proxy` when the DSN is `http`, we expect the proxy to remain free from requests
- setting the `proxy` option to `""` when the `http(s)_proxy` variables are set, should lead to the proxy to remain free from requests too